### PR TITLE
[ORPO] add the log odds part to the loss

### PR DESCRIPTION
**Pull request by kashif**
# What does this PR do?

Adds the missing log-odds part to the loss

### CODE REVIEW

File: examples/research_projects/diffusion_orpo/train_diffusion_orpo_sdxl_lora.py [hunk 1]
```diff
@@ -986,7 +986,9 @@ def compute_time_ids(original_size, crops_coords_top_left):
                 model_losses = F.mse_loss(model_pred.float(), target.float(), reduction="none")
                 model_losses = model_losses.mean(dim=list(range(1, len(model_losses.shape))))
                 model_losses_w, model_losses_l = model_losses.chunk(2)
-                log_odds = model_losses_w - model_losses_l
+                log_odds = (model_losses_w - model_losses_l) - (
+                    torch.log(1 - torch.exp(model_losses_w)) - torch.log(1 - torch.exp(model_losses_l))
```
Comment (sayakpaul) [inline examples/research_projects/diffusion_orpo/train_diffusion_orpo_sdxl_lora.py:990]: Did you validate this? Does it run well?

Comment (kashif) [reply examples/research_projects/diffusion_orpo/train_diffusion_orpo_sdxl_lora.py:990]: checking do not merge

Comment (sayakpaul) [reply examples/research_projects/diffusion_orpo/train_diffusion_orpo_sdxl_lora.py:990]: I did check and it didn't train well. But let's see.

**Comment by gokyeongryeol** [comment]
@sayakpaul  Shouldn't the log_odds be `-(model_losses_w - model_losses_l)`?
Referring the pdf of gaussian distribution, `log_prob` is akin to `-mse` not `mse`.
